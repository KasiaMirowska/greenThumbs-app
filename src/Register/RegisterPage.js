import React from 'react';
import RegisterForm from './RegisterForm';

export default class RegisterPage extends React.Component {

    handleRegister = () => {
        const {history} =this.props;
        history.push('/login');
    }

    render() {
        return (
            <div>
                <h2>REGISTER</h2>
                <RegisterForm onRegisterSuccess={this.handleRegister}/>
            </div>
        )
    }
}
