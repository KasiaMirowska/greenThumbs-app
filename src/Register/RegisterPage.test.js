import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as BR} from 'react-router-dom';
import RegisterPage from './RegisterPage';


describe('List component', () => {
    const props = {
        history: {push: '/login'}
    }
   
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><RegisterPage {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><RegisterPage {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})