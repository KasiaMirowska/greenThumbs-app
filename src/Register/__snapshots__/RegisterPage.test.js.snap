// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="auth"
>
  <div
    id="auth"
  >
    <h1>
      REGISTER 
    </h1>
    <div
      className="error"
    />
  </div>
  <div>
    <div>
      <form
        className="auth-form"
        onSubmit={[Function]}
      >
        <input
          className="form__field form__field3"
          id="fullname"
          onClick={[Function]}
          placeholder="fullname"
          type="input"
        />
        <br />
        <input
          className="form__field form__field3"
          id="username"
          onClick={[Function]}
          placeholder="username"
          type="input"
        />
        <br />
        <input
          className="form__field form__field3"
          id="password"
          onClick={[Function]}
          placeholder="password"
          type="password"
        />
        <button
          type="submit"
        >
          Submit
        </button>
        <button
          className="login-button"
          onClick={[Function]}
          type="reset"
        >
          Cancel
        </button>
      </form>
    </div>
  </div>
</div>
`;
